src/mc-fle2-payload-iev.c

```diff
@@ -84,6 +90,14 @@ mc_FLE2IndexedEncryptedValue_parse (mc_FLE2IndexedEncryptedValue_t *iev,
    CHECK_AND_RETURN (
       mc_reader_read_u8 (&reader, &iev->original_bson_type, status));
 
+   /* Check that original_bson_type is a valid bson type. */
+   if ((iev->original_bson_type < 0) || (iev->original_bson_type > 0xFF)) {
```

Comment:
I don't believe this condition can ever be met, because `original_bson_type` only has a range of `[0, 0xff]`. This will allow all 256 byte values through, but most of those are not valid type tags.